ecommerce/serializers.py

```diff
@@ -56,18 +56,23 @@ def get_courses(self, instance):
 
         model_class = instance.product.content_type.model_class()
         if model_class is CourseRun:
-            courses = [instance.product.content_object.course]
+            return [
+                CourseSerializer(
+                    instance.product.content_object.course,
+                    context={**self.context, "filter_products": True},
+                ).data
+            ]
         elif model_class is Program:
             courses = Course.objects.filter(
                 program=instance.product.content_object
             ).order_by("position_in_program")
+
+            return CourseSerializer(
+                courses, many=True, context={**self.context, "filter_products": False}
```

Comment:
The function of `filter_products`, and the reason for it being `True` for CourseRuns and `False` for Programs isn't very intuitive. Can you add some detail about that, maybe in a comment or two?